src/main/java/net/ccbluex/liquidbounce/injection/mixins/minecraft/entity/MixinClientPlayerEntity.java

```diff
@@ -241,6 +242,11 @@ private boolean hookOmnidirectionalSprintC(ClientPlayerEntity instance) {
         return isOmniWalking(instance);
     }
 
+    @Inject(method = "tickMovement", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/network/ClientPlayerEntity;canStartSprinting()Z"))
+    private void hookPostMovementInputEvent(CallbackInfo ci) {
+        EventManager.INSTANCE.callEvent(new PostMovementInputEvent(new DirectionalInput(input), input.jumping));
+    }
```

Comment:
To fix inputs not being updated based on current controls, which has now been fixed by you injecting the MovementInputEvent at the end instead of before input change.

Reply:
Perfect, then we can use that solution and remove the event. There was no reason for the MovementInputEvent being injected that early, it was a straight up bug haha